test/object-store/sync/client_reset.cpp

```diff
@@ -4144,3 +4145,717 @@ TEST_CASE("client reset with embedded object", "[client reset][local][embedded o
+                advance_and_notify(*local_realm);
+                TableRef table = get_table(*local_realm, "TopLevel");
+                REQUIRE(table->size() == 1);
+                auto obj = table->get_object(0);
+                auto col = table->get_column_key("any_mixed");
+                if (test_mode == ClientResyncMode::DiscardLocal) {
+                    List list{local_realm, obj, col};
+                    REQUIRE(list.size() == 0);
+                }
+                else {
+                    List list{local_realm, obj, col};
+                    REQUIRE(list.size() == 1);
+                    auto n_list = list.get_list(0);
+                    REQUIRE(n_list.get_any(0).get_int() == 50);
+                }
+            })
+            ->run();
+    }
+    SECTION("delete collection remotely and add locally same index.") {
+        ObjectId pk_val = ObjectId::gen();
+        SyncTestFile config2(init_sync_manager.app(), "default");
+        config2.schema = config.schema;
+        auto test_reset = reset_utils::make_fake_local_client_reset(config, config2);
+        auto test = test_reset
+                        ->setup([&](SharedRealm realm) {
+                            auto table = get_table(*realm, "TopLevel");
+                            auto obj = table->create_object_with_primary_key(pk_val);
+                            auto col = table->get_column_key("any_mixed");
+                            obj.set_collection(col, CollectionType::List);
+                            List list{realm, obj, col};
+                            list.insert_collection(0, CollectionType::List);
+                            auto n_list = list.get_list(0);
+                            n_list.insert(0, Mixed{30});
+                        })
+                        ->make_local_changes([&](SharedRealm local_realm) {
+                            TableRef table = get_table(*local_realm, "TopLevel");
+                            REQUIRE(table->size() == 1);
+                            auto obj = table->get_object(0);
+                            auto col = table->get_column_key("any_mixed");
+                            List list{local_realm, obj, col};
+                            list.insert_collection(0, CollectionType::List);
+                            auto n_list = list.get_list(0);
+                            n_list.insert(0, Mixed{50});
+                        })
+                        ->make_remote_changes([&](SharedRealm remote_realm) {
+                            TableRef table = get_table(*remote_realm, "TopLevel");
+                            REQUIRE(table->size() == 1);
+                            auto obj = table->get_object(0);
+                            auto col = table->get_column_key("any_mixed");
+                            List list{remote_realm, obj, col};
+                            REQUIRE(list.size() == 1);
+                            list.remove(0);
+                        });
+        {
```

Comment:
is this necessary? can't you chain on `->on_post_reset...` like with the other tests?